service/mantle/fedex/FedexServices.xml

```diff
@@ -0,0 +1,162 @@
+            <parameter name="responseMap"/>
+            <parameter name="labelRequestMap"/>
+        </out-parameters>
+        <actions>
+
+            <!-- Fetching Shipment Entity -->
+            <entity-find-one entity-name="mantle.shipment.Shipment" value-field="shipment" for-update="true"/>
+            <if condition="!shipment"><return error="true" message="Shipment not found with ID ${shipmentId}"/></if>
+
+            <!-- Fetching ShipmentRouteSegment Entity -->
+            <entity-find-one entity-name="mantle.shipment.ShipmentRouteSegment" value-field="routeSegment"/>
+            <if condition="!routeSegment">
+                <return error="true" message="Shipment Route Segment not found with ID ${shipmentId}:${shipmentRouteSegmentSeqId}"/>
+            </if>
+
+            <!--If shipmentPackageSeqId present, will call service for one package, else for all packages-->
+            <entity-find entity-name="mantle.shipment.ShipmentPackageRouteSeg" list="packageRouteSegList">
+                <econdition field-name="shipmentId"/><econdition field-name="shipmentRouteSegmentSeqId"/>
+                <econdition field-name="shipmentPackageSeqId" ignore-if-empty="true"/>
+            </entity-find>
+
+            <!-- Calling LabelRequestMap for all Packages -->
+            <service-call name="mantle.fedex.FedexServices.make#LabelRequestMap" in-map="context" out-map="request"/>
+            <if condition="!request.labelRequestMap"><return/></if>
+            <set field="labelRequestMap" from="request.labelRequestMap"/>
+
+            <!-- Setting Connection With Fedex -->
+            <service-call name="mantle.fedex.FedexServices.send#FedexRequest" out-map="response"
+                in-map="[shippingGatewayConfigId:shippingGatewayConfigId, method:'POST', path:'ship/v1/shipments',
+                contentType:'application/json', requestMap:request.labelRequestMap, xLocale:'en-US']"/>
+
+            <set field="responseMap" from="response.responseMap"/>
+            <if condition="!responseMap">
+                <return type="danger" message="Response not received on calling get#ShippingLabel service"></return>
+            </if>
+            <if condition="responseMap.output.transactionShipments.alerts &amp;&amp; responseMap.output.transactionShipments.alerts[0]">
+                <iterate list="responseMap.output.transactionShipments.alerts" entry="alert">
+                    <message type="danger">Fedex error(${alert.alertType}) - ${alert.code}: ${alert.message}</message>
+                </iterate>
+            </if>
+
+            <!-- Storing eachPackage's trackingCode, labelDate and labelImage in packageRouteSeg -->
+            <set field="iterator" type="Integer" value="0"/>
+            <iterate list="packageRouteSegList" entry="packageRouteSeg">
+                <set field="packageRouteSeg.trackingCode"
+                     from="responseMap.output.transactionShipments[0].pieceResponses[iterator].trackingNumber"/>
```

Comment:
I think you don't need extra variable like iterator. You can access the index using packageRouteSeg_index here. Please check.